graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1114,10 +1119,13 @@ def write_edge_structure(
             )
             .withColumnRenamed(NODE_MAPPING_INT, "src_int_id")
             .withColumnRenamed(NODE_MAPPING_STR, "src_str_id")
-            .repartition(self.num_output_files, F.col("src_str_id"))
         )
+
+        # If edge is homogeneous, we'll re-use the same mapping for both src and dst
+        if src_ntype == dst_ntype:
+            src_node_id_mapping.cache()
```

Comment:
A question here: why do we only do cache when src_ntype == dst_ntype? Whether we should always do that?

Reply:
If the edge type is heterogeneous, the node_id mapping DF will only end up being used once so there's no point in trying to cache it.

Caching makes sense in occasions where a DF is being used more than once so we want to maintain it in cluster memory.